pkg/cloud/amazon/s3_storage.go

```diff
@@ -622,8 +623,15 @@ func (s *s3Storage) List(ctx context.Context, prefix, delim string, fn cloud.Lis
 		return true
 	}
 
+	var s3Input *s3.ListObjectsInput
```

Comment:
nit: can you copy your comment in the commit into the code as well? It'll be useful to understand why we have this special case whenever someone reads this file